engines/order_management/app/services/order_management/reports/bulk_coop/bulk_coop_report.rb

```diff
@@ -158,15 +159,23 @@ def report_line_items
         end
 
         def customer_payments_total_cost(line_items)
-          line_items.map(&:order).uniq.sum(&:total)
+          unique_orders(line_items).sum(&:total)
         end
 
         def customer_payments_amount_owed(line_items)
-          line_items.map(&:order).uniq.sum(&:outstanding_balance)
+          if OpenFoodNetwork::FeatureToggle.enabled?(:customer_balance, @user)
+            unique_orders(line_items).sum(&:new_outstanding_balance)
+          else
+            unique_orders(line_items).sum(&:outstanding_balance)
+          end
         end
 
         def customer_payments_amount_paid(line_items)
-          line_items.map(&:order).uniq.sum(&:payment_total)
+          unique_orders(line_items).sum(&:payment_total)
+        end
+
+        def unique_orders(line_items)
```

Comment:
I'm not entirely sure about this optimization now. It's passing tests but I want to be sure we're not messing with the wrong list of line_items before moving on.

Reply:
Good point. You pass `line_items` in but the result can be cached from a previous call. Even if it does the right think in this context, we can't rely on it.

I think the OrderGrouper calls these methods based on the column names.
https://github.com/openfoodfoundation/openfoodnetwork/blob/0654e3d1cff3335e9d3075c03047162867b65db2/lib/open_food_network/order_grouper.rb#L63-L81

Once it's called with the items in a group and then it's also called for a summary. Those items are probably different and the summary would probably show the wrong value!

Thanks for picking that up, Pau. I wouldn't have seen it. But this is probably breaking reports. We should take this commit out. Can you write a spec for this?

Reply:
Yeah, I think I prefer to leave this out better to stickto what we have than introducing a regression or deviate too much from the goal of this PR.

Reply:
I ended up changing a bit the commit instead of removing it. It became https://github.com/openfoodfoundation/openfoodnetwork/pull/6879/commits/fec09b692e3d5b3a23a4f327de3aaf8223a81759.